packages/text-bitmap/src/BitmapText.ts

```diff
@@ -301,53 +331,199 @@ export class BitmapText extends Container
+            pageMesh.vertices[(index * 8) + 5] = yPos + (textureFrame.height * scale);
+
+            pageMesh.vertices[(index * 8) + 6] = xPos;
+            pageMesh.vertices[(index * 8) + 7] = yPos + (textureFrame.height * scale);
+
+            pageMesh.uvs[(index * 8) + 0] = textureUvs.x0;
+            pageMesh.uvs[(index * 8) + 1] = textureUvs.y0;
+
+            pageMesh.uvs[(index * 8) + 2] = textureUvs.x1;
+            pageMesh.uvs[(index * 8) + 3] = textureUvs.y1;
+
+            pageMesh.uvs[(index * 8) + 4] = textureUvs.x2;
+            pageMesh.uvs[(index * 8) + 5] = textureUvs.y2;
+
+            pageMesh.uvs[(index * 8) + 6] = textureUvs.x3;
+            pageMesh.uvs[(index * 8) + 7] = textureUvs.y3;
         }
 
         this._textWidth = maxLineWidth * scale;
         this._textHeight = (pos.y + data.lineHeight) * scale;
 
-        // apply anchor
-        if (this.anchor.x !== 0 || this.anchor.y !== 0)
+        for (const i in pagesMeshData)
         {
-            for (let i = 0; i < lenChars; i++)
+            const pageMeshData = pagesMeshData[i];
+
+            // apply anchor
+            if (this.anchor.x !== 0 || this.anchor.y !== 0)
             {
-                this._glyphs[i].x -= this._textWidth * this.anchor.x;
-                this._glyphs[i].y -= this._textHeight * this.anchor.y;
+                let vertexCount = 0;
+
+                const anchorOffsetX = this._textWidth * this.anchor.x;
+                const anchorOffsetY = this._textHeight * this.anchor.y;
+
+                for (let i = 0; i < pageMeshData.total; i++)
+                {
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetX;
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetY;
+
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetX;
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetY;
+
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetX;
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetY;
+
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetX;
+                    pageMeshData.vertices[vertexCount++] -= anchorOffsetY;
+                }
             }
+
+            this._maxLineHeight = maxLineHeight * scale;
+
+            const vertexBuffer = pageMeshData.mesh.geometry.getBuffer('aVertexPosition');
+            const textureBuffer = pageMeshData.mesh.geometry.getBuffer('aTextureCoord');
+            const indexBuffer = pageMeshData.mesh.geometry.getIndex();
+
+            vertexBuffer.data = pageMeshData.vertices;
+            textureBuffer.data = pageMeshData.uvs;
+            indexBuffer.data = pageMeshData.indices;
+
+            vertexBuffer.update();
+            textureBuffer.update();
+            indexBuffer.update();
+        }
+
+        // return back to the pool..
+        for (const i in pagesMeshData)
+        {
+            pageMeshDataPool.push(pagesMeshData[i]);
```

Comment:
If you return a mesh-data to the (shared) pool, then another BitmapText could snatch it in its updateText. Am I wrong here?i 

Let's say you have bitmap-texts B1 and B2:

1. B1.updateText() --> it returns child meshes's data to pagesMeshDataPool
2. B2.updateText() --> it grabs from pagesMeshDataPool and then `addChild` to itself

---
Again, inline comments here how it works would be helpful.

Reply:
haha! great catch, definitely something fishy there! Will tweak a bit..